subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -270,9 +271,52 @@ void ull_scan_aux_setup(memq_link_t *link, struct node_rx_hdr *rx)
 
 	if (h->adv_addr) {
 #if defined(CONFIG_BT_CTLR_SYNC_PERIODIC)
-		if (sync && (pdu->tx_addr == scan->per_scan.adv_addr_type) &&
-		    !memcmp(ptr, scan->per_scan.adv_addr, BDADDR_SIZE)) {
-			scan->per_scan.state = LL_SYNC_STATE_ADDR_MATCH;
+		/* Check if Periodic Advertising Synchronization to be created
+		 */
+		if (sync) {
```

Comment:
This adds another 40 lines to a function that's almost 500 lines long. I strongly suggest that we look into splitting this function.

Reply:
I will refactor in a separate commit in this PR

Reply:
done